src/Microsoft.Identity.Web/HttpContextExtensions.cs

```diff
@@ -24,7 +24,7 @@ internal static void StoreTokenUsedToCallWebAPI(this HttpContext httpContext, Jw
         /// </summary>
         /// <param name="httpContext">Http context associated with the current request.</param>
         /// <returns><see cref="JwtSecurityToken"/> used to call the Web API.</returns>
-        internal static JwtSecurityToken GetTokenUsedToCallWebAPI(this HttpContext httpContext)
+        internal static JwtSecurityToken? GetTokenUsedToCallWebAPI(this HttpContext httpContext)
```

Comment:
In this method (and similar ones), we expect `httpContext` to have a value here. However, if null is passed in, we get an NPE. So we either need to return null or throw.

Reply:
For public APIs, we should always check for NULLs and throw NullArgumentException. 
For internal APIs like this - use your best judgement - I think in most cases we can add a null check, but this is where you know best. If you can't decide, my advice would be throw, i.e. be stingy with returning NULLs.